mart/models/modular.py

```diff
@@ -125,9 +125,13 @@ def attack_step(self, batch, batch_idx):
     # Training
     #
     def training_step(self, batch, batch_idx):
-        # FIXME: Would be much nicer if batch was a dict!
-        input, target = batch
-        output = self(input=input, target=target, model=self.model, step="training")
+        # FIXME: Would be much nicer if batch is always a dictionary!
+        # We are going to feed the raw batch of a dictionary to self.model, but also making it backward-compatible with the tuple batch format.
+        if isinstance(batch, dict):
+            input = target = None
+        else:
+            input, target = batch
```

Comment:
Maybe better logic is:
```
input = target = None
if instance(batch, Iterable) and len(batch) == 2:
   input, target = batch
```

Reply:
Nice. But I would change `Iterable` to `Sequence` because `Iterable` does not require `__len__`.